src/libOpenImageIO/imagebufalgo_xform.cpp

```diff
@@ -307,13 +408,92 @@ warp_impl(ImageBuf& dst, const ImageBuf& src, const Imath::M33f& M,
 
 
 
+// Is the PV `option` in the `recognized` list? If so, return true.
+// Is it in the `obsolete` list? If so, return false.
+// Is it in neither? Return false.
+static bool
+IBA_find_optional(const ParamValue& option, cspan<ustring> recognized,
+                  cspan<ustring> obsolete = {})
+{
+    for (auto&& r : recognized)
+        if (option.name() == r)
+            return true;
+    for (auto&& o : obsolete)
+        if (option.name() == o)
+            return false;
+    return false;
+}
+
+
+
+// Given list of `options`, check if any are unrecognized or obsolete.
+// Return true if all are fine, false if not.
+static bool
+IBA_check_optional(ImageBufAlgo::KWArgs options, cspan<ustring> recognized,
+                   cspan<ustring> obsolete = {})
+{
+    bool ok = true;
+    for (auto&& pv : options) {
+        ok &= IBA_find_optional(pv, recognized, obsolete);
+    }
+    return ok;
+}
+
+
+
+bool
+ImageBufAlgo::warp(ImageBuf& dst, const ImageBuf& src, M33fParam M,
+                   KWArgs options, ROI roi, int nthreads)
+{
+    static ustring recognized[] = { "filtername"_us,    "filterwidth"_us,
+                                    "wrap"_us,          "edgeclamp"_us,
+                                    "recompute_roi"_us, "filterptr"_us };
+    IBA_check_optional(options, recognized);
+
+    Filter2D::ref filterref;
+    if (options.contains("filterptr", TypePointer)) {
+        auto f = options.find("filterptr", TypePointer);
+        OIIO_DASSERT(f != options.cend());
+        filterref = Filter2D::ref(f->get<const Filter2D*>(),
+                                  Filter2D::no_destroy);
+    } else {
+        filterref = get_warp_filter(options.get_string("filtername"),
```

Comment:
This spooks me that `filtername` and `filterwidth` might have a typo here or in `recognized` and it'll be hard to catch unless there's a test for each of these. Same goes for all other places this kind of usage occurs. Could we create a single const variable somewhere, such as `const ustring OIIO_Strings::filtername("filtername")` that we can instead reference? This way if there's a typo when using the variable we'd get an immediate compile time error.

Whether this variable is exposed to users as well is another question. Exposing it would lend more of the same safety to users.

Another small benefit of this approach is that you wouldn't have the overhead of having to create these ustrings each time the function is called.

Reply:
Oh, I see your point. Sure, I can do that.